pkg/upstream/cluster/cluster_manager.go

```diff
@@ -250,15 +250,31 @@ func (cm *clusterManager) RemovePrimaryCluster(clusterNames ...string) error {
 	return nil
 }
 
+
 // EDS Handler
-func NewSimpleHostHandler(c types.Cluster, hostConfigs []v2.Host) {
+func NewHealthySimpleHostHandler(c types.Cluster, hostConfigs []v2.Host) {
 	snap := c.Snapshot()
 	hosts := make([]types.Host, 0, len(hostConfigs))
+	allHostDown := true
+	oldHosts := map[string]bool{}
+
+	snap.HostSet().Range(func(host types.Host) bool {
+		oldHosts[host.AddressString()] = true
+		return true
+	})
+
 	for _, hc := range hostConfigs {
-		hosts = append(hosts, NewSimpleHost(hc, snap.ClusterInfo()))
+		host := NewSimpleHost(hc, snap.ClusterInfo())
+		hosts = append(hosts, host)
+		if host.Health() || oldHosts[host.AddressString()] != true {
```

Comment:
I'm not sure this is the right way, two questions:
1. skip updating when it is subset of the old cluster, then, how to remove a host from a old cluster?
2. host also contains some other information, eg. metadata, but we just checking the address.

Reply:
> I'm not sure this is the right way, two questions:
> 
> 1. skip updating when it is subset of the old cluster, then, how to remove a host from a old cluster?
> 2. host also contains some other information, eg. metadata, but we just checking the address.


i think we should not skip. 
Although the host to be updated is unhealthy, the host to be deleted may no longer belong to the cluster

Reply:
> > I'm not sure this is the right way, two questions:
> > 
> > 1. skip updating when it is subset of the old cluster, then, how to remove a host from a old cluster?
> > 2. host also contains some other information, eg. metadata, but we just checking the address.
> 
> i think we should not skip. Although the host to be updated is unhealthy, the host to be deleted may no longer belong to the cluster
I agree with you. The problem I have encountered is that when the number of pods is relatively small, some pod status is not updated because the associated node is down , and the actually available pods status may jitter(network instability in edge-scenario between pods and api-server).

Reply:
> > I'm not sure this is the right way, two questions:
> > 
> > 1. skip updating when it is subset of the old cluster, then, how to remove a host from a old cluster?
> > 2. host also contains some other information, eg. metadata, but we just checking the address.
> 
> i think we should not skip. Although the host to be updated is unhealthy, the host to be deleted may no longer belong to the cluster

I agree with you. The problem I have encountered is that when the number of pods is relatively small, some pod status is not updated because the associated node is down , and the actually available pods status may jitter(network instability in edge-scenario between pods and api-server).